File: namaz-native/overlay/app/src/main/java/app/namaz/tr/BootReceiver.java
package app.namaz.tr;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BootReceiver extends BroadcastReceiver {
    @Override public void onReceive(Context context, Intent intent) {
        String json = context.getSharedPreferences(PrayerScheduler.PREFS, 0)
                .getString(PrayerScheduler.KEY_SCHEDULE_JSON, null);
        if (json != null && !json.isEmpty()) {
            try { PrayerScheduler.scheduleFromJson(context, json); } catch (Exception ignored) {}
        }
        PrayerWidgetProvider.updateAll(context);
    }
}

File: namaz-v8/app/src/main/java/app/namaz/tr/v8/alarm/BootReceiver.kt
package app.namaz.tr.v8.alarm

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.work.OneTimeWorkRequestBuilder
import androidx.work.WorkManager

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action in supportedActions) {
            WorkManager.getInstance(context).enqueue(OneTimeWorkRequestBuilder<PrayerRescheduleWorker>().build())
        }
    }

    companion object {
        private val supportedActions = setOf(
            Intent.ACTION_BOOT_COMPLETED,
            Intent.ACTION_TIMEZONE_CHANGED,
            Intent.ACTION_TIME_CHANGED,
        )
    }
}
